numpy/core/src/umath/loops.c.src

```diff
@@ -763,7 +784,22 @@ BOOL_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED
         }
     }
     else {
-        if (run_binary_simd_@kind@_BOOL(args, dimensions, steps)) {
+        /*
+         * scalar operations can occur e.g. in masked arrays where a
+         * stride 0 array is used to represent no or full mask
+         */
+        if (steps[0] == 0) {
+            if (steps[1] == 0) {
+                BOOL_SCALAR_OP(1, 1, ip2, (*args[0] @OP@ *args[1]));
```

Comment:
I think we might want `const bool val = *args[0] @OP@ *args[1]`, so that it doesn't get calculated repeatedly inside the loop?